src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8706,6 +8711,17 @@ internal static bool IsRightAssociative(SyntaxKind op)
             }
         }
 
+        internal static bool IsNonAssociative(SyntaxKind op)
+        {
+            switch (op)
+            {
+                case SyntaxKind.RangeExpression:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
         enum Precedence : uint
         {
             Expression = 0, // Loosest possible precedence, used to accept all expressions
```

Comment:
range has lower precedence than add/multiply?  interesting.  so you can do:  ```1 + 1 .. 2 + 2```      and get it parsed as ```(1 + 1) .. (2 + 2)```, but if you had ```1 << 1 .. 2 << 2``` that would be parsed as ```1 << (1 .. 2) << 2```?

That seems wonky to me.  It feels like for the binary operators range would either have the greatest or lowest precedence.  So we'd parse the above two in the same way regardless of operator.  My intuition is that .. should have the loosest precedence since there is a reasonable interpretation of ```1 + 1 .. 2 + 2``` as ```(1 + 1) .. (2 + 2)```, but there isn't really a reasonable interpreation of ```1 + (1..2) + 2``` since ranges themselves will rarely interact with operators, wihle their constituent start/ends might.

Reply:
This is currently incredibly arbitrary - we're waiting on LDM to decide the precedence. I just kind of threw it somewhere without thinking about it (it's really really easy to change - just shuffle this enum).

Thanks for the feedback though :)

(I think I said "uh, let's copy Rust, I guess", and then I promptly mis-remembered what precedence Rust's range has 👍 🎉)

Reply:
Out of curiosity, what is rust's precedence here?  If there's general consensus among languages that offer this, it would probably make sense to lean toward prior art :)

Reply:
Weaker than everything except assignments. `a + b .. c` is `(a + b) .. c`, as is `1 << 4 .. 100`. It's a little weird, since convention is to remove spaces (`a + b..c`) - nearly all Rust code I've seen in the wild explicitly parenthesizes anyway (`(a + b)..c`).

https://doc.rust-lang.org/book/second-edition/appendix-02-operators.html

![image](https://user-images.githubusercontent.com/953151/33570307-3de777d8-d8e1-11e7-9e1a-4bf3ce7b1de0.png)

Reply:
> Weaker than everything except assignments

Weaker than truly everything?  Or just weaker than all binaries?  For example, if i have this:

```a ? b : c .. d```

is that:

```(a ? b : c) .. d``` or ```a ? b : (c .. d)```.  My gut reading is that it's the latter.

Reply:
Oh, i guess rust has no ternaries.  So they don't have to deal with that question.  My point above stands though on what i think the precedence should be for C#.

Reply:
Yeah, Rust's ternary is just its if statement (which is an expression) - `let x = if a { b } else { c };`. I would agree that it makes sense to be the latter.